activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -592,6 +592,65 @@ def rewhere(conditions)
       unscope(where: conditions.keys).where(conditions)
     end
 
+    # Returns a new relation, which is the logical union of this relation and the one passed as an
+    # argument.
+    #
+    # The two relations must be structurally compatible: they must be scoping the same model, and
+    # they must differ only by +where+ (if no +group+ has been defined) or +having+ (if a +group+ is
+    # present). Neither relation may have a +limit+, +offset+, or +uniq+ set.
+    #
+    #    Post.where("id = 1").or(Post.where("id = 2"))
+    #    # SELECT `posts`.* FROM `posts`  WHERE (('id = 1' OR 'id = 2'))
+    #
+    def or(other)
+      spawn.or!(other)
+    end
+
+    def or!(other)
+      combining = group_values.any? ? :having : :where
+
+      unless structurally_compatible?(other, combining)
+        raise ArgumentError, 'Relation passed to #or must be structurally compatible'
+      end
+
+      unless other.is_a?(NullRelation)
+        left_values = send("#{combining}_values")
+        right_values = other.send("#{combining}_values")
+
+        common = left_values & right_values
+        mine = left_values - common
+        theirs = right_values - common
```

Comment:
in your example @sgrif if `a != 1` wouldn't the first fail after one check whereas the second would still make two checks?